b2luigi/batch/processes/gbasf2.py

```diff
@@ -0,0 +1,764 @@
+        os.makedirs(self.log_file_dir, exist_ok=True)
+
+        self.dirac_user = get_dirac_user(self.gbasf2_env)
+        #: Maximum number of times that each job in the project can be rescheduled until the project is declared as failed.
+        self.max_retries = get_setting("gbasf2_max_retries", default=0, task=self.task)
+
+        #: Store number of times each job had been rescheduled
+        self.n_retries_by_job = Counter()
+
+        #: Local storage for ``n_retries_by_job`` counter
+        # so that it persists even if luigi process is killed and restarted.
+        # TODO: Maybe a small database (e.g. tindydb/pickledb) would be more appropriate
+        self.retries_file_path = os.path.join(self.log_file_dir, "n_retries_by_grid_job.json")
+        if os.path.isfile(self.retries_file_path):
+            with open(self.retries_file_path, "r") as retries_file:
+                retries_from_file = json.load(retries_file)
+                self.n_retries_by_job.update(retries_from_file)
+
+        # Store dictionary with n_jobs_by_status in attribute to check if it changed,
+        # useful for printing job status on change only
+        self._n_jobs_by_status = ""
+
+        # Store whether the job had already been successful in a variable b/c
+        # there's actions we want to do only the first time that
+        # ``get_job_status`` returns a success.
+        self._project_had_been_successful = False
+
+    def get_job_status(self):
+        """
+        Get overall status of the gbasf2 project.
+
+        First obtain the status of all (sub-) jobs in a gbasf2 project, similar
+        to ``gb2_job_status``, and return an overall project status, e.g. when
+        all jobs are done, return ``JobStatus.successful`` to show that the
+        gbasf2 project succeeded.
+
+        The status of each individual job can be one of::
+
+            [Submitting, Submitted, Received, Checking, Staging, Waiting, Matched, Rescheduled,
+             Running, Stalled, Completing, Done, Completed, Failed, Deleted, Killed]
+
+        (Taken from  https://github.com/DIRACGrid/DIRAC/blob/rel-v7r1/WorkloadManagementSystem/Client/JobStatus.py)
+
+        """
+        # If project is does not exist on grid yet, so can't query for gbasf2 project status
+        if not check_project_exists(self.gbasf2_project_name, self.gbasf2_env, dirac_user=self.dirac_user):
+            raise RuntimeError(
+                f"Could not find any jobs for project {self.gbasf2_project_name} on the grid.\n" +
+                "Probably there was an error during the project submission when running the gbasf2 command.\n" +
+                "Try if you can run the gbasf2 command used manually in a terminal with gbasf2 set up:\n" +
+                " ".join(self._build_gbasf2_submit_command())
+            )
+
+        job_status_dict = get_gbasf2_project_job_status_dict(self.gbasf2_project_name, self.gbasf2_env, dirac_user=self.dirac_user)
```

Comment:
As I do not know `gbasf2`: can you gain speed by asking for all projects in a single call and then caching it for the next seconds? If yes, you can use the `BatchJobStatusCache`.